Add biocore, biosets, and biofit

psmyth94 opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

psmyth94 committed:
add biocore recipe generated from grayskull

psmyth94 committed:
add my id

psmyth94 committed:
Merge branch 'main' into biocore

psmyth94 committed:
added packaging as dependency

psmyth94 committed:
Merge branch 'biocore' of github.com:psmyth94/staged-recipes into biocore

psmyth94 committed:
include both biosets and biocore recipes

psmyth94 commented:
I confirm that I am willing to be listed as a maintainer for the biocore and biosets package.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/biosets/meta.yaml```, ```recipes/biocore/meta.yaml```) and found it was in an excellent condition.
I do have some suggestions for making it better though...


For **recipes/biosets/meta.yaml**:

 * No valid build backend found for Python recipe for package `biosets` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.

For **recipes/biocore/meta.yaml**:

 * No valid build backend found for Python recipe for package `biocore` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.

psmyth94 commented:
@conda-forge/help-python, ready for review!

psmyth94 committed:
Merge branch 'main' into add-biocore-and-biosets

psmyth94 committed:
Merge branch 'main' into add-biocore-and-biosets

psmyth94 committed:
Merge branch 'main' into add-biocore-and-biosets

psmyth94 committed:
Merge branch 'main' into add-biocore-and-biosets

psmyth94 committed:
chore: update biocore, biosets and add biofit recipes

Updated the biocore recipe to version 1.1.1 and biosets recipe to
version 1.2.1 in meta.yaml. Added a new recipe for biofit version
0.0.1.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/biocore/meta.yaml```, ```recipes/biofit/meta.yaml```, ```recipes/biosets/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/biocore/meta.yaml**:

 * ℹ️ No valid build backend found for Python recipe for package `biocore` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.
 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).

For **recipes/biofit/meta.yaml**:

 * ❌ There are 1 too many lines.  There should be one empty line at the end of the file.

For **recipes/biofit/meta.yaml**:

 * ℹ️ No valid build backend found for Python recipe for package `biofit` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.
 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).

For **recipes/biosets/meta.yaml**:

 * ℹ️ No valid build backend found for Python recipe for package `biosets` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.
 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).

psmyth94 committed:
feat: update python version constraints in meta.yaml

Updated the python version constraints in the meta.yaml files for
biocore, biofit, and biosets recipes. Added python_min and python_max
variables to set the minimum and maximum supported python versions.
Also added setuptools to the host requirements.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/biocore/meta.yaml```, ```recipes/biofit/meta.yaml```, ```recipes/biosets/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/biocore/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).

For **recipes/biofit/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).

For **recipes/biosets/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of the recipe, you should usually use `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).

psmyth94 committed:
feat: add python version constraints to meta.yaml

Added python version constraints to the meta.yaml files for biocore,
biofit, and biosets recipes. Included python_min variable to specify
the minimum supported python version in the test requirements.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/biosets/meta.yaml```, ```recipes/biofit/meta.yaml```, ```recipes/biocore/meta.yaml```) and found it was in an excellent condition.

psmyth94 committed:
fix(biofit): add pyarrow-hotfix to requirements

Added pyarrow-hotfix to the requirements section in the meta.yaml file
to address compatibility issues with the latest version of pyarrow.

shermansiu commented:
1. Why are we manually setting the Python min and max variables at the top? They should already be set by default.
2. I'm not sure if we can add several packages at once, even if they're related. I'm sure that someone else can clarify this issue.
